tools/submission/submission-checker.py

```diff
@@ -0,0 +1,554 @@
+"""
+A checker for mlperf inference submissions
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import json
+import logging
+import os
+import re
+import sys
+
+# pylint: disable=missing-docstring
+
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger("main")
+
+
+MODEL_CONFIG = {
+    "v0.5": {
+        "models": ["ssd-small", "ssd-large", "mobilenet", "resnet", "gnmt"],
+        "required-scenarios-datacenter": {
+            # anything goes
+        },
+        "required-scenarios-edge": {
+            # anything goes
+        },
+        "accuracy-target": {
+            "mobilenet": ("acc", 71.68 * 0.98),
+            "resnet": ("acc", 76.46 * 0.99),
+            "ssd-small": ("mAP", 22 * 0.99),
+            "ssd-large": ("mAP", 20 * 0.99),
+            "gnmt": ("bleu", 23.9 * 0.99),
+        },
+        "performance-sample-count": {
+            "mobilenet": 1024,
+            "resnet": 1024,
+            "ssd-small": 256,
+            "ssd-large": 64,
+            "gnmt": 3903900,
+        },
+        "seeds": {
+            "qsl_rng_seed": 3133965575612453542,
+            "sample_index_rng_seed": 665484352860916858,
+            "schedule_rng_seed": 3622009729038561421,
+        },
+    },
+    "v0.7": {
+        "models": ["ssd-large", "resnet", "rnnt", "3d-unet", "dlrm", "bert"],
+        "required-scenarios-datacenter": {
```

Comment:
I think in additional to "required", we also need to check the optional and not-allowed ones. For example, if anyone submits Server scenario results for Edge, the checker should complain

Reply:
done